src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -3197,6 +3197,11 @@ public function set_attribute( $name, $value ): bool {
 			 * @see https://html.spec.whatwg.org/#attributes-3
 			 */
 			$escaped_new_value = in_array( $comparable_name, wp_kses_uri_attributes() ) ? esc_url( $value ) : esc_attr( $value );
+
+			if ( '' === $escaped_new_value ) {
+				return false;
+			}
```

Comment:
this looks _almost_ complete! what happens if someone supplies `''` as the new value? Maybe we need to check to make sure that `$escaped_new_value !== $value && '' === $escaped_new_value`?

Reply:
Thanks for pointing it out. I have updated it.